From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: local-build <local-build@localhost>
Date: Mon, 16 Jun 2026 04:00:00 +0800
Subject: [PATCH] arm64: dts: qcom: gaokun3: force GSI mode for SPI
 touchscreen

Add qcom,force-gsi-mode property to the spi6 node to force the
Qualcomm GENI SPI controller into GSI DMA mode for the touchscreen.

---
 arch/arm64/boot/dts/qcom/sc8280xp-huawei-gaokun3.dts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/qcom/sc8280xp-huawei-gaokun3.dts b/arch/arm64/boot/dts/qcom/sc8280xp-huawei-gaokun3.dts
index 1234567..abcdefg 100644
--- a/arch/arm64/boot/dts/qcom/sc8280xp-huawei-gaokun3.dts
+++ b/arch/arm64/boot/dts/qcom/sc8280xp-huawei-gaokun3.dts
@@ -642,6 +642,7 @@ &i2c14 {
 &spi6 {
 	pinctrl-0 = <&spi6_default>;
 	pinctrl-names = "default";
+	qcom,force-gsi-mode;
 
 	status = "okay";
 
-- 
2.53.0
